mintpy/prep_aria.py

```diff
@@ -1,216 +1,290 @@
+    parser.add_argument('-w','--work-dir', dest='workDir', type=str, default="./",
+                        help='The working directory for MintPy.')
+    parser.add_argument('-s','--stack-dir', dest='stackDir', type=str, required=True,
+                        help='The directory which contains stack VRT files.')
+    parser.add_argument('-u','--unwrap-stack-name', dest='unwStack', type=str,
                         default="unwrapStack.vrt",
                         help='Name of the stack VRT file of unwrapped data.\n'+
-                              ' default: unwrapStack.vrt')
-    parser.add_argument('-c', '--coherence-stack-name', dest='cohStack', type=str,
+                             'default: unwrapStack.vrt')
+    parser.add_argument('-c','--coherence-stack-name', dest='cohStack', type=str,
                         default="cohStack.vrt",
                         help='Name of the stack VRT file of coherence data.\n'+
-                              'default: cohStack.vrt')
-    parser.add_argument('-l', '--conn-comp-name', dest='connCompStack', type=str,
+                             'default: cohStack.vrt')
+    parser.add_argument('-l','--conn-comp-name', dest='connCompStack', type=str,
                         default="connCompStack.vrt",
                         help='Name of the stack VRT file of connected component data.\n' +
-                              'default: connCompStack.vrt')
-    parser.add_argument('-i', '--incidence-angle', dest='incidenceAngle', type=str,
-                        required=True,
+                             'default: connCompStack.vrt')
+    parser.add_argument('-i','--incidence-angle', dest='incAngle', type=str, required=True,
                         help='Name of the incidence angle file')
-    parser.add_argument('-d', '--dem', dest='dem', type=str,
-                        required=True,
+    parser.add_argument('-d','--dem', dest='dem', type=str, required=True,
                         help='Name of the DEM file')
 
     return parser
 
+
 def cmd_line_parse(iargs = None):
     parser = create_parser()
     inps = parser.parse_args(args=iargs)
-
+    inps.stackDir = os.path.abspath(inps.stackDir)
     return inps
 
+
+####################################################################################
 def extract_metadata(stack):
 
+    print('extract metadata from {}'.format(stack))
     ds = gdal.Open(stack, gdal.GA_ReadOnly)
+    metaUnw = ds.GetRasterBand(1).GetMetadata("unwrappedPhase")
 
+    # copy over all metadata from unwrappedPhase
     meta = {}
+    for key, value in metaUnw.items():
+        if key not in ["Dates", "perpendicularBaseline"]:
+            meta[key] = value
+
+    meta["ANTENNA_SIDE"] = -1
     meta["PROCESSOR"] = "isce"
     meta["FILE_LENGTH"] = ds.RasterYSize
     meta["LENGTH"] = ds.RasterYSize
-    meta["ORBIT_DIRECTION"] = "DESCENDING"
+    meta["ORBIT_DIRECTION"] = "DESCENDING"    #check with ARIA-tools team
```

Comment:
Is there a way to extract this parameter, instead of hardwiring a fixed one? I have not looked into the stack vrt files in details yet.

Reply:
I was looking into that and it is not in VRT's. I think it is good to add that information to the VRT files.

Reply:
https://github.com/aria-tools/ARIA-tools/pull/75